### PR TITLE
feature: multiple connections

### PR DESCRIPTION
**Pull request by wedgwood**
This PR is for the feature multi-connections support(#10).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,22 +13,25 @@ function fastifyPostgres (fastify, options, next) {
...
+  fastify.pg[name] = {
     connect: pool.connect.bind(pool),
     pool: pool,
     Client: pg.Client,
     query: pool.query.bind(pool)
-  })
+  }
 
-  fastify.addHook('onClose', onClose)
+  fastify.addHook('onClose', (fastify, done) => pool.end(done))
```
Comment (delvedor) [inline index.js]: Why this change?
The reason of the separated function is that if something goes wrong its easier understand which function has failed, since it has a name ;)

Comment (wedgwood) [reply index.js]: 1. I had tried `name` configurable first time, but I think maybe it makes some user confuesed. sometimes `pg` is an connection container and sometimes it is an connection. I can not make a choice myself. If you think `name` configuable better, I could do it.

2. using closure because I need refer the connection pool. If I still use `onClose` function, I cloud not get the pool to end.

File: index.js [hunk 2]
```diff
@@ -13,22 +13,34 @@ function fastifyPostgres (fastify, options, next) {
...
+    if (!fastify.pg) {
+      fastify.decorate('pg', {})
+    }
 
-  next()
-}
+    fastify.pg[name] = db
+  } else {
+    if (fastify.pg) {
+      console.warn('fastify-postgres has already registered')
```
Comment (mcollina) [inline index.js]: we should error if it was already register.

File: test.js [hunk 3]
```diff
@@ -30,12 +51,13 @@ test('should be able to connect and perform a query', t => {
...
 
   fastify.register(fastifyPostgres, {
+    name: 'test',
     connectionString: 'postgres://postgres@localhost/postgres'
   })
 
   fastify.ready(err => {
     t.error(err)
-    fastify.pg.connect(onConnect)
+    fastify.pg.test.connect(onConnect)
```
Comment (mcollina) [inline test.js]: I would not change the tests, and keep using the non-prefixed version. But add a new test to check on the prefixed version.

File: index.js (hunk 2 above)
Comment (wedgwood) [reply index.js]: got it

File: test.js (hunk 3 above)
Comment (wedgwood) [reply test.js]: got it

**Comment by mcollina** [comment]
LGTM. @delvedor do you want to have another look before releasing?
